indico/modules/rb/models/rooms.py

```diff
@@ -696,22 +699,32 @@ def is_owned_by(self, user):
         return user in GroupProxy.get_named_default_group(manager_group)
 
     @hybrid_method
-    def is_in_digest_window(self, exclude_first_day=False):
+    def is_in_digest_window(self, reservation, exclude_first_day=False):
```

Comment:
I really dislike the fact that the regular and expression version of this hybrid method are different - also, the expression one will most likely retrieve the cross product between Room and Reservation when used in a query where Room and Reservation are not explicitly joined.

However, when I was with @DavidAndreev we couldn't find a cleaner solution for this quickly. Or at least none that wouldn't require rewriting the logic to handle these notifications (which, honestly, might not be too bad - even with unit tests the current notification logic is not that maintainable and will probably cause someone headaches in the future).